app/src/main/java/com/awscherb/cardkeeper/data/handler/BaseRealmHandler.java

```diff
@@ -4,67 +4,50 @@
 
 import java.util.List;
 
+import io.reactivex.Completable;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import io.reactivex.disposables.Disposables;
+import io.reactivex.functions.Consumer;
 import io.realm.Realm;
+import io.realm.RealmChangeListener;
+import io.realm.RealmConfiguration;
 import io.realm.RealmObject;
 import io.realm.exceptions.RealmException;
-import rx.Emitter;
-import rx.Observable;
-import rx.functions.Action1;
 
 abstract class BaseRealmHandler<T extends RealmObject & BaseModel> extends BaseHandler<T> {
 
     public abstract Class<T> getModelClass();
 
-    private Observable<T> doAsync(T object, Realm realm, Action1<T> action1) {
-        return Observable.fromEmitter(e -> {
-                    realm.beginTransaction();
-                    try {
-                        action1.call(object);
-                        realm.commitTransaction();
-                    } catch (RuntimeException exception) {
-                        realm.cancelTransaction();
-                        e.onError(new RealmException("Error during transaction.", exception));
-                    } catch (Error error) {
-                        realm.cancelTransaction();
-                        e.onError(error);
-                    } finally {
-                        realm.close();
-                    }
-                    e.onNext(object);
-                    e.onCompleted();
-                },
-                Emitter.BackpressureMode.BUFFER);
-    }
-
     //================================================================================
     // Get
     //================================================================================
 
     @Override
-    public Observable<T> getObject(long id) {
-        Realm realm = Realm.getDefaultInstance();
-        return realm.where(getModelClass()).equalTo("id", id).findAll()
-                .asObservable()
-                .flatMapIterable(object -> object)
+    public Single<T> getObject(long id) {
+        final Realm realm = Realm.getDefaultInstance();
+        return singleReam(realm)
+                .map(r -> r.where(getModelClass()).equalTo("id", id).findAll())
```

Comment:
"id" string probably should be a some const. Even though this seems to be a project wide policy, refactoring this will be problematic later on.